azure/converters/rules.go

```diff
@@ -32,7 +32,7 @@ func SecurityRuleToSDK(rule infrav1.SecurityRule) network.SecurityRule {
 			SourcePortRange:          rule.SourcePorts,
 			DestinationAddressPrefix: rule.Destination,
 			DestinationPortRange:     rule.DestinationPorts,
-			Access:                   network.SecurityRuleAccessAllow,
+			Access:                   network.SecurityRuleAccess(rule.Action),
```

Comment:
Does this assume that `rule.Action` will always be set and is a required field? I don't see any webhook defaulting or defaulting via kubebuilder so I was just wondering.

Reply:
+1, in order for this to be a non-breaking change we could make the field optional and default to `Allow`

Reply:
+1 @willie-yao , good find! Yep. I should be defaulting Action to Allow.
I will add ` +kubebuilder:default=Basic` to the types and also make the field optional.

Reply:
Updated via 21cedcd497b5ce604a52fb2935c8a73d4dcb5607